spec/models/user_spec.rb

```diff
@@ -0,0 +1,130 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe "Validations" do
+    subject{FactoryBot.create :user}
+
+    context "Valid attribute" do
+      it{is_expected.to be_valid}
+    end
+
+    context "Name invalid presence" do
+      it{should validate_presence_of(:name)}
+    end
+
+    context "Name invalid min length" do
+      it{should validate_length_of(:name).is_at_least(3)}
+    end
+
+    context "Name invalid max length" do
+      it{should validate_length_of(:name).is_at_most(250)}
+    end
+
+    context "Email invalid presence" do
+      it{should validate_presence_of(:email)}
+    end
+
+    context "Email invalid max length" do
+      it{should validate_length_of(:email).is_at_most(250)}
+    end
+
+    context "Email invalid format" do
+      it{should allow_value("doanphua4@gmail.com").for(:email)}
+    end
+
+    context "Email invalid uniqueness" do
+      it{should validate_uniqueness_of(:email).ignoring_case_sensitivity}
+    end
+
+    context "Password invalid presence" do
+      it{should have_secure_password}
+    end
+
+    context "Password invalid min lenth" do
+      it{should validate_length_of(:password).is_at_least(6)}
+    end
+
+    context "Password invalid max length" do
+      it{should validate_length_of(:password).is_at_most(20)}
+    end
+  end
+
+  describe "Instance mothods" do
+    describe "#remember" do
+      context "Remember success" do
+        before{subject.remember}
+        it{expect(subject.remember_digest).not_to be_nil}
+      end
+    end
+
+    describe "#forget" do
+      context "Forget success" do
+        before{subject.forget}
+        it{expect(subject.remember_digest).to be_nil}
+      end
+    end
+
+    describe "#authenticated?" do
+      context "Authenticated true" do
+        before{subject.remember}
+        it{expect(subject.authenticated? "remember", subject.remember_token).to be true}
```

Comment:
dùng subject ở ngoài context
bên trong it e dùng is_expected nhé e

Reply:
vâng ạ.